### DISCUSSION TITLE
Failed to build opencv-python

### DISCUSSION THREAD

**Discussion by der-manu** [theyosh/TerrariumPI#671] [Q&A] [answered]
I can not build opencv-python
running on an Raspberry 4 (with 1GB Ram)
I tried to give it more swap and also connected a SSD via USB3 in case the SD Card (sandisk) causes this error.
I am a bit clueless how do i get the dependencies to build and run TerrariumPI.
If one of you has an idea, please let me know

build/lib.linux-armv7l-3.7/numpy/core/_multiarray_umath.cpython-37m-arm-linux-gnueabihf.so" failed with exit status 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for numpy
Failed to build opencv-python numpy
ERROR: Could not build wheels for opencv-python, numpy, which is required to install pyproject.toml-based projects
(venv) root@raspberrypi:/mnt/TerrariumPI#

**Comment by theyosh** [comment]
Which Raspberry PI OS did you install?

Also, do not install it as user root. https://theyosh.github.io/TerrariumPI/install/

**Reply by der-manu** [reply]
I started with RPi Pi OS lite legacy. After your suggestions -don't install as root- I deleted the directory and started new. But i get to the same point, it does not build opencv due missing dependencies/ problems while build.
I ordered a new "original" power supply in case that forces this error.

**Reply by theyosh** [reply]
I do not think the power supply is the issue.. The strange thing is that it should not compile opencv. It should use a precompiled version: https://www.piwheels.org/project/opencv-python-headless/

Did you run the installer like: `sudo ./install.sh`

As I am just able to install the software without any issues. I do not know what is causing your issue...

**Reply by der-manu** [reply]
I restarted three times from scratch, 2/3 times I started with raspios_oldstable_lite_armhf-2022-01-28 image, the 3rd time I used 
 raspios_oldstable_lite_armhf-2021-12-02.
When I start from a vanilla image I noticed that I dont have bc and python3-venv installed, and - for some reason - the installer script does not install those packages.
At 3rd try I manually installed "python3-pip python3-dev python3-venv libopenexr23 libilmbase23 liblapack3 libatlas3-base bc screen git watchdog i2c-tools pigpio sqlite3 ffmpeg sispmctl ntp libxslt1.1 libglib2.0-dev" just like the installer script should do and then I ran the install.sh script (as user pi) just like you adviced.

unfortunately I get to the same point 
![Screenshot_2022-03-08_15-30-05](https://user-images.githubusercontent.com/6383246/157258306-b3db1f3c-41ee-418e-b0b1-cf07387075f1.png)
The attempt-counter counts up, but package does not get installed.
I really do not now why I have such problems, your instructions are pretty good.
I can not explain why the install.sh does not install the dependencies.
Maybe you could give me a hint.

**Reply by theyosh** [reply]
I think I fixed it. Do a `git pull` and rerun the installer: `sudo ./install.sh` and I think it should work now.

**Reply by der-manu** [reply]
Thanks. Since the latest update it now works, just as described.
Thank you for providing this awsome project.

**Comment by theyosh** [comment, marked as answer]
Fixed
